### PR TITLE
avoid double saving of replacements nodes, removed replacement_cache, test

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: include/libtorrent/kademlia/node.hpp [hunk 1]
```diff
@@ -162,8 +162,10 @@ class TORRENT_EXTRA_EXPORT node : boost::noncopyable
 	// bucket is not full.
 	void add_node(udp::endpoint const& node);
 
+#ifndef TORRENT_NO_DEPRECATE
```
Comment (arvidn) [inline include/libtorrent/kademlia/node.hpp]: if this is not actually exposed in the API, I imagine you could just remove it, no?

**Comment by arvidn** [comment]
good catch! I would imagine you could just remove that code though

File: include/libtorrent/kademlia/node.hpp (hunk 1 above)
Comment (aldenml) [inline include/libtorrent/kademlia/node.hpp]: yes, agree, I will remove it...removing code :)

**Comment by arvidn** [comment]
back-ported here: https://github.com/arvidn/libtorrent/pull/1661

**Pull request merged**